pkg/registry/common/dnsresolve/common.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnsresolve
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+)
+
+// Domain represents wanted domain and service
+type Domain struct {
```

Comment:
Why are we combining the domain and service here?  Also wouldn't that no longer be the *domain* ?  Why not just pass the network service name "service@domain" ?

Reply:
@edwarnicke Service here is not network service, it is service to lookup SRV record :)

Reply:
Could you say more, I don't quite follow.

Reply:
https://github.com/networkservicemesh/sdk/pull/371/files#diff-c9d8b841fcd143230a5ca28860c6e096R41

Reply:
Take a look at service parameter of SRV record format https://en.wikipedia.org/wiki/SRV_record

Reply:
In k8s Service is `Service.Spec.Ports.Name` field. For example in k8s example of service can be `svc.proxy.registry` and so on.

Reply:
OK... I'm not quite following why we are passing this through the context.  We can extract the domain from the Connection.NetworkServiceName directly... and we should probably pick a common SRV.Service name... what were your thoughts there?

Reply:
>nd we should probably pick a common SRV.Service name... what were your thoughts there?

I think yes, we can add some constant for SRV.Service name for registries. In this case, the struct Domain will be deleted.

Reply:
@edwarnicke should we put SRC.Service name for registries to API?

Reply:
Question... why couldn't it simply be a constant in the resolver package?

Reply:
I thought that it can be used by API users, finally fixed.